homeassistant/components/device_tracker/__init__.py

```diff
@@ -95,30 +103,43 @@ def setup(hass, config):
     """Setup device tracker."""
     yaml_path = hass.config.path(YAML_DEVICES)
 
-    conf = config.get(DOMAIN, {})
+    conf = CONFIG_SCHEMA(config).get(DOMAIN, [])
 
-    # Config can be an empty list. In that case, substitute a dict
-    if isinstance(conf, list):
-        conf = conf[0] if len(conf) > 0 else {}
+    def _get_conf_value(key_name, default):
```

Comment:
I don't like this approach. It is already confusing enough that a config entry for 1 platform applies to all platforms. I feel that by limiting it to just the first platform it is at least less confusing. Eventually we should get a better way to specify device tracker not-platform-related config.